docs/source/getting_started.rst

```diff
@@ -1,31 +1,198 @@
+
+   def p() -> None:
+       print('hello')
+
+   a = p()   # Type check error: p has None return value
+
+Arguments with default values can be annotated as follows:
+
+.. code-block:: python
+
+   def greeting(name: str, prefix: str = 'Mr.') -> str:
+      return 'Hello, {} {}'.format(name, prefix)
+
+Installing mypy
+***************
 
 Mypy requires Python 3.4 or later to run.  Once you've
 `installed Python 3 <https://www.python.org/downloads/>`_,
 you can install mypy with:
 
 .. code-block:: text
 
-    $ python3 -m pip install mypy
+    python3 -m pip install mypy
 
 Note that even though you need Python 3 to run ``mypy``, type checking
 Python 2 code is fully supported, as discussed in :ref:`python2`.
 
-Installing from source
-**********************
+Running mypy
+************
+
+You can type check a program by using the ``mypy`` tool, which is
+basically a linter -- it checks your program for errors without actually
+running it::
 
-To install mypy from source, clone the
-`mypy repository on GitHub <https://github.com/python/mypy>`_ and then run
-``pip install`` locally:
+   $ mypy program.py
 
-.. code-block:: text
+All errors reported by mypy are essentially warnings that you are free
+to ignore, if you so wish.
+
+More command line options are documented in :ref:`command-line`.
+
+.. note::
+
+   Depending on how mypy is configured, you may have to run mypy like
+   this::
+
+     $ python3 -m mypy program.py
+
+The typing module
+*****************
+
+The ``typing`` module contains many definitions that are useful in
+statically typed code. You typically use ``from ... import`` to import
+them (we'll explain ``Iterable`` later in this document):
+
+.. code-block:: python
+
+   from typing import Iterable
+
+   def greet_all(names: Iterable[str]) -> None:
+       for name in names:
+           print('Hello, {}'.format(name))
+
+For brevity, we often omit the ``typing`` import in code examples, but
+mypy will give an error if you use definitions such as ``Iterable``
+without first importing them.
+
+Mixing dynamic and static typing
+********************************
+
+Mixing dynamic and static typing within a single file is often
+useful. For example, if you are migrating existing Python code to
+static typing, it may be easiest to do this incrementally, such as by
+migrating a few functions at a time. Also, when prototyping a new
+feature, you may decide to first implement the relevant code using
+dynamic typing and only add type signatures later, when the code is
+more stable.
+
+.. code-block:: python
+
+   def f():
+       1 + 'x'  # No static type error (dynamically typed)
+
+   def g() -> None:
+       1 + 'x'  # Type check error (statically typed)
+
+.. note::
+
+   The earlier stages of mypy, known as the semantic analysis, may
+   report errors even for dynamically typed functions. However, you
+   should not rely on this, as this may change in the future.
+
+.. _stubs-intro:
+
+Library stubs and the Typeshed repo
```

Comment:
This section emphasizes creating your own stubs. IMO it would be better if we emphasized the *existence* of stubs (and sometimes the lack of stubs!) and reminded people that if they get complaints about a call (or import, etc.) to some stdlib or 3rd party module they should look in typeshed to see what the stub actually says. This seems much more important than linking to how you can create your own stubs.